### ISSUE TITLE
bug(ipeps): regularized_eigh backward has wrong sign in _regularized_eigh_bwd

### ISSUE THREAD

**Issue by yingjerkao** [tenax-lab/tenax#316]
## Summary

`src/tenax/algorithms/ad_utils.py::regularized_eigh` ships the **negative of the correct projector gradient**. Every production AD path that uses this custom VJP is affected, including the 1-site C4v explicit-AD recommended path and any 2-site shared-tensor path that routes through \`_compute_projector_tensor\`'s tracer branch.

Discovered while reviewing plan PR #315 / Task 7 (\`feat/lorentzian-projector-backward\`) where a parallel Lorentzian backward was being wired in — the new path matches FD, the existing one does not.

## Reproducer

```python
import jax, jax.numpy as jnp
from tenax.algorithms.ad_utils import regularized_eigh

jax.config.update(\"jax_enable_x64\", True)

key = jax.random.PRNGKey(0)
C = jax.random.normal(key, (6, 6))
rho = C @ C.T  # PSD, n=6
M = jnp.eye(6)  # any symmetric observable

def L(r):
    eigvals, eigvecs = regularized_eigh(r)
    k = 3
    P = eigvecs[:, -k:][:, ::-1]
    return jnp.trace(P.T @ M @ P)

# FD gradient along symmetric direction
eps = 1e-5
h = 0.5 * (jax.random.normal(key, (6, 6)) + jax.random.normal(key, (6, 6)).T)
fd = (L(rho + eps * h) - L(rho - eps * h)) / (2 * eps)

# AD gradient contracted with h
ad = jnp.sum(jax.grad(L)(rho) * h)

print(f\"FD:  {fd:+.10f}\")
print(f\"AD:  {ad:+.10f}\")
print(f\"ratio AD/FD = {ad / fd}\")  # expect +1.0, actually gives -1.0
```

## Root cause

\`_regularized_eigh_bwd\` at \`src/tenax/algorithms/ad_utils.py:389–395\` constructs

\`\`\`python
diff = w[:, None] - w[None, :]
# ...
F = diff / (diff**2 + eps**2)   # F_ij = 1 / (w_i - w_j)
M_bar = inner * F               # WRONG sign
\`\`\`

JAX reverse-mode convention for symmetric eigendecomposition wants \`F_ij = 1 / (w_j - w_i)\` here. The fix is either:
- swap the subtraction: \`diff = w[None, :] - w[:, None]\`, or
- transpose the contraction: \`M_bar = inner * (-F)\` / \`F.T\`.

Single sign flip; one-liner fix.

## Why nothing has blown up

\`regularized_eigh\` is used only at the CTM projector construction site, where the projector columns feed into a Frobenius-normalized / Gram-reduced CTM sweep. The wrong-sign gradient has been covered for by:
- Phase / sigma gauge tricks that reshape the effective loss landscape,
- Loose CTM tolerances that wash out projector-column sensitivity,
- Line searches accepting the wrong descent direction via positional luck or the L-BFGS Hessian history flipping sign once per reset.

At χ=8 on the 2-site shared-tensor C4v AFM Heisenberg these tricks no longer cover for it, which is exactly the symptom in issue #299 (E stalls at ~−0.558 vs literature −0.6548). Plan PR #315's Task 7 routes around \`regularized_eigh\` entirely via the new \`truncated_eigh_regularized\` in \`_lorentzian_eigh.py\` and closes the gap (diagnostic: E=−0.66022 at χ=8, beating literature χ=16 D=2).

## Evidence table

| Path | \`L1 = tr(Pᵀ M P)\` directional deriv | \`L2 = ‖Pᵀ target‖²\` directional deriv |
|---|---|---|
| FD (truth) | +1.56265686 | −1.51668923 |
| \`jnp.linalg.eigh\` (native) | +1.56265686 ✓ | matches ✓ |
| \`regularized_eigh\` (current) | **−1.56265686 ✗** | **+1.51668924 ✗** |
| \`truncated_eigh_regularized\` (plan Task 2) | +1.56265686 ✓ | matches ✓ |

Both losses are gauge-invariant (quadratic in \`P\`, so \`L(P) = L(-P)\`), so the column-sign freedom is not in play.

## Fix plan (separate PR — do not bundle with PR for plan #315)

1. Fix the sign flip in \`_regularized_eigh_bwd\`.
2. Add a FD-AD regression test on a gauge-invariant structured loss (e.g. \`tr(Pᵀ M P)\`) to \`tests/test_ad_utils.py\` — currently there is no such coverage.
3. Re-run the 1-site C4v χ=16 benchmark and confirm the recommended path's measured energy is unchanged or improved (gauge tricks should still converge, and may converge faster with a correct gradient).
4. Audit any other consumers of \`regularized_eigh\` — grep for imports across the codebase.
5. Close this issue and reference the linked issue #299 for the convergence-gap symptom.

## Relationship to plan #315 / PR for Task 7

Task 7's \`projector_backward=\"lorentzian\"\` path is unaffected — it does not go through \`regularized_eigh\`. The fix for this issue is orthogonal and should be merged first if possible, to reduce the scope of the plan PR. If the fix lands later, the plan PR's Tasks 2–8 still stand on their own merits (adding a truncation-correction kernel as an opt-in).

cc @tenax-lab
